DOC-1740: RC Quick start for Prometheus/Grafana (#2270)
RedisLabs/redislabs-docs: content/rc/cloud-integrations/prometheus-integration.md

---
Title: Prometheus integration with Redis Cloud
linkTitle: Prometheus integration
description: To collect and display metrics data from your databases, you can connect your Prometheus or Grafana server to your Redis Cloud databases.
weight: $weight
alwaysopen: false
categories: ["RC"]
---

To collect and display metrics data from your databases, you can connect your Prometheus or Grafana server to your Redis Cloud subscription.

- [Prometheus](https://prometheus.io/) is an open-source systems monitoring and alerting toolkit that can scrape metrics from different sources.

> - [Grafana](https://grafana.com/) is an open-source, feature-rich metrics dashboard and graph editor that can process Prometheus data.

Reviewer: ```suggestion
- [Grafana](https://grafana.com/) is an open-source metrics dashboard and graph editor that can process Prometheus data.
```

Reviewer 2: I would clarify the opening a little: ... you can connect Prometheus to the 'Prometheus compatible endpoint' in the Redis Cloud subscription, then you can connect Grafana to Prometheus for visualization.  You can't connect Grafana directly to the RC metrics endpoint and you technically don't need Grafana, it's just a very common pattern to add Grafana for visualization.

> - [Prometheus](https://prometheus.io/) is an open-source systems monitoring and alerting toolkit that can scrape metrics from different sources.
> - [Grafana](https://grafana.com/) is an open-source, feature-rich metrics dashboard and graph editor that can process Prometheus data.

Reviewer 3: Our style guide spells "open source" without hyphens, even when used as an adjective.
```suggestion
- [Prometheus](https://prometheus.io/) is an open source systems monitoring and alerting toolkit that can scrape metrics from different sources.
- [Grafana](https://grafana.com/) is an open source, feature-rich metrics dashboard and graph editor that can process Prometheus data.
```

> For more information on how Prometheus communicates with Redis Enterprise Software clusters, see [Prometheus integration with Redis Enterprise Software]({{< relref "/rs/clusters/monitoring/prometheus-integration" >}}).

Reviewer 3: So does Prometheus work the same for both Software and Cloud? If so, maybe just shorten "Redis Enterprise Software clusters" to "Redis Enterprise clusters" to imply that the info applies to both Cloud and Software.
```suggestion
For more information on how Prometheus communicates with Redis Enterprise clusters, see [Prometheus integration with Redis Enterprise Software]({{< relref "/rs/clusters/monitoring/prometheus-integration" >}}).
```

{{< note >}}

> You must [enable VPC peering]({{< relref "/rc/security/vpc-peering" >}}) for your subscription to connect Prometheus and Grafana to your Redis Cloud databases. VPC peering is available only with Flexible or Annual subscriptions.  Prometheus and Grafana cannot connect to databases on Fixed or Free subscriptions.

Reviewer: Might consider moving this below the prereq list.

{{< /note >}}

> ## Quick start

Reviewer: I would suggest breaking this up with some level 3 headings: 

- Prerequisites
- Create Prometheus configuration file
-  Set up your Prometheus and Grafana servers
- Verify connection (you can probably come up with a better heading for this)
- Configure the Grafana datasource
- Add dashboards for cluster, node, and database metrics.
-

> To get started with custom monitoring with Prometheus on Docker:

Reviewer: ```suggestion
To get started monitoring with Prometheus on Docker:
```

> 1. To connect Prometheus and Grafana to your Redis Cloud databases, they must be running on an machine that:

Reviewer: This is more of a prerequisite than a step. I would add a prereq section.

Reviewer: Might be better to add the note about VPC peering after the prereqs since it's mentioned here.

> 1. To connect Prometheus and Grafana to your Redis Cloud databases, they must be running on an machine that:

Reviewer 3: I agree with Kaitlyn's suggestion to add a prerequisites section. Also, you might want to reword this a bit to avoid starting with "to" twice in a row and also avoid phrases like "be running".
```suggestion
1. To connect Prometheus and Grafana to your Redis Cloud databases, they must be running on a machine that:
```

    - Is running on the same cloud provider as your Redis Cloud subscription
    - Is located in the same region as your Redis Cloud subscription
    - Is connected to the VPC subnet that is peered with your Redis Cloud subscription

>     - Allows outbound connections to port 8070, so that Prometheus can scrape the Redis Cloud server for data

Reviewer: ```suggestion
    - Allows outbound connections to port 8070, so Prometheus can scrape the Redis Cloud server for data
```
Taking out "that" is one of the easiest ways to reduce the length of your sentences.

    - Allows inbound connections to port 9090 for Prometheus and 3000 for Grafana

>     Create a directory on this machine called `prometheus`.

Reviewer: Is this a step? It could easily be combined with the next step. By telling them to create a directory and then create a new file in that directory.

1. Within that directory, create a file called `prometheus.yml`.

> 1. Add the following contents to `prometheus.yml`. Replace `<instance_ip>` with the IP address of the machine and `<cluster_name>` with the FQDN portion of your database endpoint.

Reviewer: ```suggestion
1. Add the following contents to `prometheus.yml`. Replace `<instance_ip>` with the IP address of the machine and `<cluster_name>` with the FQDN portion of your database endpoint.
```
I'm not super familiar with the cloud UI, is FQDN supplied for them. We usually define it on first use in an article but if it's just listed as FQDN in the UI this should be fine.

Reviewer: I would also love if we added the purpose of this file. Is it a config file?

        #rule_files:
        # - "first.rules"
        # - "second.rules"

        scrape_configs:
        # scrape Prometheus itself
        - job_name: prometheus
            scrape_interval: 10s
            scrape_timeout: 5s
            static_configs:
                - targets: ["<instance_ip>:9090"]

        # scrape Redis Cloud
        - job_name: redis-cloud
            scrape_interval: 30s
            scrape_timeout: 30s
            metrics_path: /
            scheme: https
            static_configs:
                - targets: ["internal.<cluster_name>:8070"]
    ```
1. Set up your Prometheus and Grafana servers.

    To set up Prometheus and Grafana on Docker containers:
     1. Create a _docker-compose.yml_ file:

        ```yml
        version: '3'
        services:
            prometheus-server:
                image: prom/prometheus
                ports:
                    - 9090:9090
                volumes:
                    - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml

            grafana-ui:
                image: grafana/grafana
                ports:
                    - 3000:3000
                environment:

>                     - GF_SECURITY_ADMIN_PASSWORD=secret

Reviewer 3: Does the user need to replace "secret" with their own password here? If so, maybe you can mention that in the instructions so they don't miss it.

                links:
                    - prometheus-server:prometheus
        ```

    1. To start the containers, run:

        ```sh
        $ docker compose up -d
        ```

>     1. To check that all the containers are up, run: `docker ps`

Reviewer: 1. Run `docker ps` to check if all the containers are up and running. 

Whenever possible, try to start action steps with a verb. 
Are they looking for a certain status in the command output such as "running" or "started." If so we should mention that here.

    1. In your browser, sign in to Prometheus at http://<instance_ip>:9090 to make sure the server is running.

>     1. Select **Status** and then **Targets** to check that Prometheus is collecting data from the Redis Enterprise cluster.
> 
>         {{<image filename="images/rs/prometheus-target.png" alt="The Redis Enterprise target showing that Prometheus is connected to the Redis Enterprise Cluster.">}}{{< /image >}}

Reviewer 3: ```suggestion
    1. Select **Status** and then **Targets** to verify that Prometheus is collecting data from the Redis Cloud cluster.

        {{<image filename="images/rs/prometheus-target.png" alt="The Redis Cloud target showing that Prometheus is connected to the Redis Cloud cluster.">}}{{< /image >}}
```

>         If Prometheus is connected to the cluster, you can type **node_up** in the Expression field on the Prometheus home page to see the cluster metrics.

Reviewer 3: ```suggestion
        If Prometheus is connected to the cluster, you can enter `node_up` in the **Expression** field on the Prometheus home page to see the cluster metrics.
```

> 1. Configure the Grafana datasource: 

Reviewer: Might consider making this a section heading instead of a step. Breaks up the page a little more to make it easier to scan.

> 1. Configure the Grafana datasource: 

Reviewer 3: ```suggestion
1. Configure the Grafana data source: 
```

>     1. Sign in to Grafana. If you installed Grafana with Docker, go to http://<instance_ip>:3000 and sign in with:

Reviewer: ```suggestion
    1. Sign in to Grafana. 
        If you installed Grafana with Docker, go to http://<instance_ip>:3000 and sign in with:
```

        - Username: admin
        - Password: secret

    1. In the Grafana configuration menu, select **Data Sources**.

    1. Select **Add data source**.

    1. Select **Prometheus** from the list of data source types.

        {{<image filename="images/rs/prometheus-datasource.png" alt="The Prometheus data source in the list of data sources on Grafana.">}}{{< /image >}}

    1. Enter the Prometheus information:

        - Name: `redis-enterprise`
        - URL: `http://<your prometheus address>:9090`
        - Access: `Server`

        {{<image filename="images/rs/prometheus-connection.png" alt="The Prometheus connection form in Grafana.">}}{{< /image >}}

    {{< note >}}

- If the network port is not accessible to the Grafana server, select the **Browser** option from the Access menu.
- In a testing environment, you can select **Skip TLS verification**.

    {{< /note >}}

1. Add dashboards for cluster, node, and database metrics.
    To add preconfigured dashboards:
    1. In the Grafana dashboards menu, select **Manage**.

>     1. Click **Import**.

Reviewer 3: ```suggestion
    1. Select **Import**.
```

>     1. Copy one of the configurations into the **Paste JSON** field.

Reviewer 3: ```suggestion
    1. Copy and enter one of the following configuration files into the **Paste JSON** field.
```

        {{% expand "database.json" %}}

```json
{{% embed-code "/rs/database.json" %}}
```

        {{% /expand%}}
        {{%expand "node.json" %}}

```json
{{% embed-code "/rs/node.json" %}}
```

        {{% /expand%}}
        {{%expand "cluster.json" %}}

```json
{{% embed-code "/rs/cluster.json" %}}
```

        {{% /expand%}}

>     1. In the Import options, select the `redis-enterprise` datasource and click **Import**.

Reviewer 3: ```suggestion
    1. In the **Import** options, select the `redis-enterprise` data source and then select **Import**.
```

> The dashboards that you create from the configurations are sample dashboards.

Reviewer: ```suggestion
The dashboards you create from the configurations are sample dashboards.
```

> The dashboards that you create from the configurations are sample dashboards.

Reviewer 3: ```suggestion
The provided configuration files create sample dashboards.
```

Reviewer 4: Alternatively: "These examples create sample dashboards."

> For more information about configuring dashboards, see the [Grafana documentation](http://docs.grafana.org).

Reviewer 3: It seems like this link redirects to another URL. Since this looks like an old link that uses http instead of https, maybe it's worth updating to the new link?

---
Title: Prometheus integration with Redis Cloud
linkTitle: Prometheus integration

> description: To collect and display metrics data from your databases, you can connect your Prometheus or Grafana server to your Redis Cloud databases.

Reviewer 2: Just to be technically accurate/clear: ... you can connect Prometheus to the 'Prometheus compatible endpoint' in the Redis Cloud subscription, then you can connect Grafana to Prometheus for visualization.  You can't connect Grafana directly to the RC metrics endpoint and you technically don't need Grafana, it's just a very common pattern to add Grafana for visualization.
